### DISCUSSION TITLE
Add mechanism to censor login passwords

### DISCUSSION THREAD

**Discussion by NotAFile** [piqueserver/piqueserver#695] [Feature Requests]
https://github.com/piqueserver/piqueserver/pull/511 exposed that we currently save passwords in e.g. the /login command in our logs

**Comment by 1AmYF** [comment]
There needs to remain a way to trace back activity of password owners in logs (in case of abusive moderation for example). Currently there is no other way to be sure than checking whose password was used for login.

**Comment by godwhoa** [comment]
How about this. It censors but lets you traceback to which password was used.
```
2019-08-20T14:27:23+0530 [piqueserver.core_commands.social#info] <Deuce> /login **********
2019-08-20T14:27:23+0530 [piqueserver.core_commands.social#info] Deuce logged in using admin[0]
```

```toml
[passwords]
admin = ["adminpass1", "adminpass2"]
```

**Comment by 1AmYF** [comment]
>Deuce logged in using admin[0]

That would be great. Also improves log readability, before you couldn't see directly which role was used for the login. Maybe for people not familiar with indexes, using a text like `logged in using admin password no. 1` or `logged in as admin #1` might be a bit more clear.

**Comment by xtreme8000** [comment]
hashing?
> Deuce: /login ****
> Deuce logged in as admin, using d41d8cd98f00b204e9800998ecf8427e

**Comment by NotAFile** [comment]
@xtreme8000 that would only help if you had some easy way of mapping the hash back to the user
